File: Problems/0069. (E) Sqrt(x)/Python3/solution.py
class Solution:
  def mySqrt(self, x: int) -> int:
    # 1 <= x <= 2**31-1, so we can just iterate
    rt = 1
    sq = 1
    while x >= sq:
      rt += 1
      sq = rt**2

    return rt-1

File: Problems/0069. (E) Sqrt(x)/Java/solution.java
class Solution {
  public int mySqrt(int x) {
    // Newton's method, need to modify Python solution to avoid casting to long

    int start=1, end=x, floorRes=0;
    while (start <= end) {
      int mid = start + (end-start) / 2;

      if (mid <= x / mid ) {
        floorRes = mid;
        start = mid+1;
      }
      else {
        end = mid-1;
      }
    } 
    return floorRes;
  }
}
